hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/SparkHoodieTableFileIndex.scala

```diff
@@ -319,6 +319,8 @@ class SparkHoodieTableFileIndex(spark: SparkSession,
    * @return relative partition path and a flag to indicate if the path is complete (i.e., not a prefix)
    */
   private def composeRelativePartitionPath(staticPartitionColumnNameValuePairs: Seq[(String, (String, Any))]): String = {
+    // NOTE: Do not blindly use default value if tableConfig does not have the hiveStylePartitioningEnable config
+    checkState(metaClient.getTableConfig.getHiveStylePartitioningEnableOrNull != null)
```

Comment:
Can you explain why it could not hold a default value? Can you check how it is initialized when table is initialized from scratch? Can you also add a test case if necessary.

Reply:
We are still investigating this issue on why hoodie.properties does not have the `hoodie.datasource.write.hive_style_partitioning` config key.

However, we are seeing that some tables do not have this configuration. 

Given that there's an unsafe usage and unsafe assumption, i figured that it would be better to add a check here to prevent worse-case scenario from happening, i.e. wrong partitioning format is assumed and a partition is then created.

Reply:
Okay, let's make the root cause clear first.